src/grpc/mod.rs

```diff
@@ -122,6 +126,67 @@ impl KeyBrokerService for KeyBroker {
         };
         Result::Ok(Response::new(reply))
     }
+
+    async fn get_online_secret(
+        &self,
+        request: Request<OnlineSecretRequest>,
+    ) -> Result<Response<OnlineSecretResponse>, Status> {
+        let r = request.into_inner();
+        let client_id = Uuid::parse_str(&r.client_id)
+            .map_err(|e| Status::internal(format!("Malformed Launch ID: {}", e)))?;
+
+        let (connection, key) = db::get_connection(client_id)
+            .map_err(|e| Status::internal(format!("Connection not found: {}", e)))?;
+
+        let mut secret_request = request::SecretRequest::new();
+
+        secret_request
+            .parse_requests(&r.secret_requests)
+            .map_err(|e| Status::internal(format!("Bad secret request: {}", e)))?;
+
+        let policies = secret_request.policies();
+
+        // Validate connection against policies
+        for p in policies {
+            p.verify(&connection)
+                .map_err(|e| Status::internal(format!("Policy validation failed: {}", e)))?;
+        }
+        info!(
+            "Policy validated succesfully. Connection: {:?}",
+            &connection
+        );
+
+        let mut secret_request = request::SecretRequest::new();
+
+        secret_request
+            .parse_requests(&r.secret_requests)
+            .map_err(|e| Status::internal(format!("Bad secret request: {}", e)))?;
+
+        let policies = secret_request.policies();
+
+        // Validate connection against policies
+        for p in policies {
+            p.verify(&connection)
+                .map_err(|e| Status::internal(format!("Policy validation failed: {}", e)))?;
+        }
+        info!(
+            "Policy validated succesfully. Connection: {:?}",
+            &connection
+        );
+
+        // this gives us the ovmf secret table, which we don't want
```

Comment:
is this comment relevant here? I thought that `payload_simple()` doesn't return the OVMF-table.